### ISSUE TITLE
Failed to connect to the proxy via bolt framework

### ISSUE THREAD

**Issue by sabari219** [slackapi/bolt-python#898]
Am trying to create a slackbot using bolt framework using proxy

#### The `slack_bolt` version

slack-bolt==1.15.1
slack-sdk==3.19.1

#### Python runtime version

Python 3.10.7

#### OS info

ProductName:    macOS
ProductVersion: 12.6.3
BuildVersion:   21G419

#### Steps to reproduce:

(Share the commands to run, source code, and project settings (e.g., setup.py))

```python
#!/usr/bin/env python3

from typing import Optional
from slack_bolt import App
import os
from dotenv import load_dotenv
import logging
import json
import xlsxwriter
from jsonpath_ng.ext import parse
import sys, getopt
import datetime
import time
from datetime import datetime, timedelta, date
import numpy as np
import pandas as pd
import re
from globalvariables import *
from makeapicall import *
from keymaker import *
from botautoreply import *
from loggerconfig import *
import ssl
import certifi
from slack_bolt import App
from slack_bolt.adapter.socket_mode import SocketModeHandler
from slack_bolt.error import BoltUnhandledRequestError
from slack_bolt.response import BoltResponse
from slack_sdk import WebClient

ssl_context = ssl.create_default_context(cafile=certifi.where())
os.environ["SLACK_HTTP_PROXY"] = "http://xxx:80"
os.environ["SLACK_APP_TOKEN"] = SLACK_APP_TOKEN
os.environ["SLACK_BOT_TOKEN"] = SLACK_BOT_TOKEN
os.environ["SLACK_SIGNING_SECRET"] = "xxx"

load_dotenv()
logger.debug("app object creation")

app_token = os.environ.get("SLACK_APP_TOKEN")
bot_token = os.environ.get("SLACK_BOT_TOKEN")
http_proxy = os.environ.get("SLACK_HTTP_PROXY")
signing_secret = os.environ.get("SLACK_SIGNING_SECRET")

client = WebClient(
    token=bot_token,
    proxy=http_proxy,
)

app = App(
    client=client,
    token=bot_token,
    ssl_check_enabled=False,
    signing_secret=signing_secret,
    raise_error_for_unhandled_request=True,
)

if __name__ == "__main__":
    handler = SocketModeHandler(app, app_token, proxy=http_proxy)
    handler.start()
```

### Expected result:

Able to connect the slack server successfully

### Actual result:

```
[2023-05-08 20:22:33,068]:client.py:79:connect_to_new_endpoint:INFO:Connected to a new endpoint...
[2023-05-08 20:22:43,069]:connection.py:281:check_state:DEBUG:This connection is already closed. (session id: 05b0383e-d95a-41dc-bdc6-5065c758a7a3)
[2023-05-08 20:22:43,069]:client.py:281:_monitor_current_session:INFO:The session seems to be already closed. Reconnecting... (session id: 05b0383e-d95a-41dc-bdc6-5065c758a7a3)
[2023-05-08 20:22:43,069]:client.py:76:connect_to_new_endpoint:INFO:Connecting to a new endpoint...
[2023-05-08 20:22:43,070]:base_client.py:252:_urllib_api_call:DEBUG:Sending a request - url: https://www.slack.com/api/apps.connections.open, query_params: {}, body_params: {}, files: {}, json_body: None, headers: {'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': '(redacted)', 'User-Agent': 'Python/3.10.11 slackclient/3.19.1 Linux/5.15.0-60-generic'}
[2023-05-08 20:22:43,342]:base_client.py:520:_perform_urllib_http_request_internal:DEBUG:Received the following response - status: 200, headers: {'date': 'Mon, 08 May 2023 20:22:43 GMT', 'server': 'Apache', 'vary': 'Accept-Encoding', 'x-slack-req-id': '04de57d5d280afac1b1694295036e041', 'x-content-type-options': 'nosniff', 'x-xss-protection': '0', 'x-robots-tag': 'noindex,nofollow', 'pragma': 'no-cache', 'cache-control': 'private, no-cache, no-store, must-revalidate', 'expires': 'Sat, 26 Jul 1997 05:00:00 GMT', 'content-type': 'application/json; charset=utf-8', 'x-accepted-oauth-scopes': 'connections:write', 'x-oauth-scopes': 'connections:write', 'access-control-expose-headers': 'x-slack-req-id, retry-after', 'access-control-allow-headers': 'slack-route, x-slack-version-ts, x-b3-traceid, x-b3-spanid, x-b3-parentspanid, x-b3-sampled, x-b3-flags', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'x-slack-unique-id': 'ZFlaE1ft4K1ULRFnnNoGrgAAADA', 'x-slack-backend': 'r', 'referrer-policy': 'no-referrer', 'access-control-allow-origin': '*', 'x-powered-by': 'HHVM/4.153.1-slack-bionic-blessed-build-565', 'content-length': '173', 'via': '1.1 slack-prod.tinyspeck.com, envoy-www-iad-gryq, envoy-edge-iad-vmym', 'x-envoy-upstream-service-time': '15', 'x-backend': 'main_normal main_canary_with_overflow main_control_with_overflow', 'x-server': 'slack-www-hhvm-main-iad-gqcz', 'x-slack-shared-secret-outcome': 'no-match', 'x-edge-backend': 'envoy-www', 'x-slack-edge-shared-secret-outcome': 'no-match', 'connection': 'close'}, body: {"ok":true,"url":"wss:\/\/wss-primary.slack.com\/link\/?ticket=3b9bb3a6-54d7-4791-bc67-102f0c571265&app_id=68b0d93818f16bdf3b84858a5f6216361bf075b9ad4a164abdd793fe17666b2f"}
[2023-05-08 20:22:43,365]:connection.py:179:connect:ERROR:Failed to establish a connection (session id: fdf862ff-3fc2-42ec-b2cc-6367fd0af238, error: Failed to connect to the proxy (proxy: http://xxx:80, connect status code: 403))
[2023-05-08 20:22:43,366]:client.py:246:_on_error:ERROR:on_error invoked (session id: 05b0383e-d95a-41dc-bdc6-5065c758a7a3, error: Exception, message: Failed to connect to the proxy (proxy: http://xxxx:80, connect status code: 403))
[2023-05-08 20:22:43,366]:connection.py:195:disconnect:INFO:The connection has been closed (session id: fdf862ff-3fc2-42ec-b2cc-6367fd0af238)
[2023-05-08 20:22:43,366]:connection.py:195:disconnect:INFO:The connection has been closed (session id: 05b0383e-d95a-41dc-bdc6-5065c758a7a3)
[2023-05-08 20:22:43,366]:client.py:191:connect:INFO:A new session has been established (session id: fdf862ff-3fc2-42ec-b2cc-6367fd0af238)
[2023-05-08 20:22:43,366]:client.py:79:connect_to_new_endpoint:INFO:Connected to a new endpoint.
```

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

**Comment by seratch** [comment]
Hi @sabari219, can you try the same with this example? https://github.com/slackapi/bolt-python/blob/main/examples/socket_mode_proxy.py

If the code fails with the same error, I am afraid that your proxy server may not allow WebSocket connections, which are required for using Socket Mode. According to the logs you've shared, an HTTP request to `https://www.slack.com/api/apps.connections.open` is successful, but establishing a WebSocket connection in the following step.

**Comment by sabari219** [comment]
After following the [code](https://github.com/slackapi/bolt-python/blob/main/examples/socket_mode_proxy.py), am getting below error.

[2023-05-09 23:07:12,431]:client.py:79:connect_to_new_endpoint:INFO:Connected to a new endpoint...
[2023-05-09 23:07:22,432]:connection.py:281:check_state:DEBUG:This connection is already closed. (session id: 7a5b2533-4e54-43c7-804b-76e8353588e3)
[2023-05-09 23:07:22,432]:client.py:281:_monitor_current_session:INFO:The session seems to be already closed. Reconnecting... (session id: 7a5b2533-4e54-43c7-804b-76e8353588e3)
[2023-05-09 23:07:22,432]:client.py:76:connect_to_new_endpoint:INFO:Connecting to a new endpoint...
[2023-05-09 23:07:22,433]:base_client.py:252:_urllib_api_call:DEBUG:Sending a request - url: https://www.slack.com/api/apps.connections.open, query_params: {}, body_params: {}, files: {}, json_body: None, headers: {'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': '(redacted)', 'User-Agent': 'Python/3.10.11 slackclient/3.19.1 Linux/5.15.0-60-generic'}
[2023-05-09 23:07:22,811]:base_client.py:520:_perform_urllib_http_request_internal:DEBUG:Received the following response - status: 200, headers: {'date': 'Tue, 09 May 2023 23:07:22 GMT', 'server': 'Apache', 'vary': 'Accept-Encoding', 'x-slack-req-id': '0979a21bb29472e7fbf98426e1fa411f', 'x-content-type-options': 'nosniff', 'x-xss-protection': '0', 'x-robots-tag': 'noindex,nofollow', 'pragma': 'no-cache', 'cache-control': 'private, no-cache, no-store, must-revalidate', 'expires': 'Sat, 26 Jul 1997 05:00:00 GMT', 'content-type': 'application/json; charset=utf-8', 'x-accepted-oauth-scopes': 'connections:write', 'x-oauth-scopes': 'connections:write', 'access-control-expose-headers': 'x-slack-req-id, retry-after', 'access-control-allow-headers': 'slack-route, x-slack-version-ts, x-b3-traceid, x-b3-spanid, x-b3-parentspanid, x-b3-sampled, x-b3-flags', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'x-slack-unique-id': 'ZFrSKptFEdmMbhkhmgM28AAAEDw', 'x-slack-backend': 'r', 'referrer-policy': 'no-referrer', 'access-control-allow-origin': '*', 'x-powered-by': 'HHVM/4.153.1-slack-bionic-blessed-build-585', 'content-length': '173', 'via': '1.1 slack-prod.tinyspeck.com, envoy-www-iad-voui, envoy-edge-iad-oisi', 'x-envoy-upstream-service-time': '18', 'x-backend': 'main_normal main_canary_with_overflow main_control_with_overflow', 'x-server': 'slack-www-hhvm-main-iad-nuit', 'x-slack-shared-secret-outcome': 'no-match', 'x-edge-backend': 'envoy-www', 'x-slack-edge-shared-secret-outcome': 'no-match', 'connection': 'close'}, body: {"ok":true,"url":"wss:\/\/wss-primary.slack.com\/link\/?ticket=259ca32c-2fe3-4c15-a116-f88b777fe2ad&app_id=68b0d93818f16bdf3b84858a5f6216361bf075b9ad4a164abdd793fe17666b2f"}
[2023-05-09 23:07:22,846]:connection.py:179:connect:ERROR:Failed to establish a connection (session id: 802c9dda-71eb-4117-9476-e226265cfe8b, error: Failed to connect to the proxy (proxy: http://ccg01proxy-vip.ccg01.phx.paypalinc.com:80, connect status code: 403))
[2023-05-09 23:07:22,846]:client.py:246:_on_error:ERROR:on_error invoked (session id: 7a5b2533-4e54-43c7-804b-76e8353588e3, error: Exception, message: Failed to connect to the proxy (proxy: http://ccg01proxy-vip.ccg01.phx.paypalinc.com:80, connect status code: 403))
[2023-05-09 23:07:22,847]:connection.py:195:disconnect:INFO:The connection has been closed (session id: 802c9dda-71eb-4117-9476-e226265cfe8b)
[2023-05-09 23:07:22,847]:connection.py:195:disconnect:INFO:The connection has been closed (session id: 7a5b2533-4e54-43c7-804b-76e8353588e3)
[2023-05-09 23:07:22,847]:client.py:191:connect:INFO:A new session has been established (session id: 802c9dda-71eb-4117-9476-e226265cfe8b)
[2023-05-09 23:07:22,847]:client.py:79:connect_to_new_endpoint:INFO:Connected to a new endpoint...
[2023-05-09 23:07:32,848]:connection.py:281:check_state:DEBUG:This connection is already closed. (session id: 802c9dda-71eb-

**Comment by seratch** [comment]
Thanks for sharing that. As you can see in the logs, your WSS (Secure WebSocket) connection towards `wss-primary.slack.com` seems to be disallowed by the proxy.
